src/rpc/mining.cpp

```diff
@@ -801,7 +802,11 @@ static RPCHelpMan getblocktemplate()
         {
             MillisecondsDouble checktxtime{std::chrono::minutes(1)};
             while (tip == hashWatchedChain && IsRPCRunning()) {
-                tip = miner.waitTipChanged(hashWatchedChain, checktxtime).hash;
+                std::optional<BlockRef> maybe_tip{miner.waitTipChanged(hashWatchedChain, checktxtime)};
+                if (!maybe_tip) {
+                    throw JSONRPCError(RPC_CLIENT_NOT_CONNECTED, "Shutting down");
+                };
+                tip = CHECK_NONFATAL(maybe_tip)->hash;
                 // Timeout: Check transactions for update
```

Comment:
Why is comment saying "Timeout: ..." when we could be here if the tip changed (no timeout)? Is the comment wrong?